internal/http/services/dataprovider/dataprovider.go

```diff
@@ -103,24 +96,83 @@ func (s *svc) Handler() http.Handler {
 	return s.handler
 }
 
-func (s *svc) setHandler() {
-	s.handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		switch r.Method {
-		case "HEAD":
-			addCorsHeader(w)
-			w.WriteHeader(http.StatusOK)
-			return
-		case "GET":
-			s.doGet(w, r)
-			return
-		case "PUT":
-			s.doPut(w, r)
-			return
-		default:
-			w.WriteHeader(http.StatusNotImplemented)
-			return
+// Composable is the interface that a struct needs to implement to be composable by this composer
+type Composable interface {
+	UseIn(composer *tusd.StoreComposer)
+}
+
+func (s *svc) setHandler() (err error) {
+	composable, ok := s.storage.(Composable)
+	if ok {
+		// A storage backend for tusd may consist of multiple different parts which
+		// handle upload creation, locking, termination and so on. The composer is a
+		// place where all those separated pieces are joined together. In this example
+		// we only use the file store but you may plug in multiple.
+		composer := tusd.NewStoreComposer()
+
+		// let the composable storage tell tus which extensions it supports
+		composable.UseIn(composer)
+
+		config := tusd.Config{
+			BasePath:      s.conf.Prefix,
+			StoreComposer: composer,
+			//Logger:        logger, // TODO use logger
+		}
+
+		handler, err := tusd.NewUnroutedHandler(config)
+		if err != nil {
+			return err
 		}
-	})
+
+		s.handler = handler.Middleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			log := appctx.GetLogger(r.Context())
+			log.Info().Msgf("tusd routing: path=%s", r.URL.Path)
+
+			switch r.Method {
+			// old fashioned download.
+
+			// GET is not part of the tus.io protocol
+			// currently there is no way to GET an upload that is in progress
+			// TODO allow range based get requests? that end before the current offset
+			case "GET":
```

Comment:
GET needs to be there to download normal files, not necessary ongoing uploads.

Reply:
which is what the current code does.